cli/bpmetadata/types_ui.go

```diff
@@ -0,0 +1,124 @@
+package bpmetadata
+
+// BlueprintUI is the top-level structure for holding UI specific metadata
+type BlueprintUI struct {
+	// The top-level input section that defines the list of properties and
+	// their sections on the deployment page
+	Input BlueprintUIInput `yaml:"input"`
+
+	// The top-level section for listing runtime (or solution output) information
+	// i.e. the console URL for the VM or a button to ssh into the VM etc based on
+	Runtime BlueprintUIOutput `yaml:"runtime"`
+
+	// The solution version corresponding to the version in the
+	// standard metadata file for keeping data in sync.
+	Version string `yaml:"version"`
+}
+
+// BlueprintUIInput is the structure for holding Input and Input Section (i.e. groups) specific metadata
+type BlueprintUIInput struct {
+
+	// Properties is a map defining all inputs on the UI
+	DisplayVariables map[string]DisplayVariable `yaml:"variables"`
+
+	// Sections is a generic structure for grouping inputs together
+	DisplaySections []DisplaySection `yaml:"sections"`
+}
+
+// Additional display specific metadata pertaining to a particular
+// input variable.
+type DisplayVariable struct {
+	// The variable name from the corresponding standard metadata file.
+	Name string `yaml:"name"`
+
+	// Visible title for the variable on the UI.
+	Title bool `yaml:"title,omitempty"`
+
+	// A flag to hide or show the property on the UI.
+	Visible bool `yaml:"visible,omitempty"`
+
+	// Property tooltip.
+	Tooltip string `yaml:"tooltip,omitempty"`
+
+	// Placeholder text (when there is no default).
+	Placeholder string `yaml:"placeholder,omitempty"`
+
+	// Text describing the validation rules for the property. Typically shown
+	// after an invalid input.
+	Validation string `yaml:"validation,omitempty"`
```

Comment:
What is an example value for this? I see some validations like pattern/max/min etc as other fields.

Reply:
This is specifically for validating Regex for text fields and ranges for int fields. Min/max is for list type fields and pattern was a dupe.

Reply:
Done.